### PR TITLE
add markdown report support for test scenario runner

### PR DESCRIPTION
**Pull request by zhenglaizhang**
pls see https://github.com/Azure/oav/pull/624#issuecomment-852687105 as an test run in my local env. note the file link is not clickable

### CODE REVIEW

**Comment by zhenglaizhang** [comment]
<h3>Azure API Test Report</h3>
  <details>
    <summary>
      <code>
        ❌
      </code>
      <b>
        galleries_0: 0 Fatals, 1 Errors
      </b>
    </summary>
    <br />
    <ul>
      <li>
        Duration:
        49.004 seconds
      </li>
      <li>
        Swagger Files
        <ul>
            <li>
              <a href="Microsoft.Compute/preview/2020-09-30/gallery.json">
                Microsoft.Compute/preview/2020-09-30/gallery.json
              </a>
            </li>
        </ul>
      </li>
      <li>
        Run Id: 202106021125-3l03r
      </li>
    </ul>
    <b>
      API Test Steps:
    </b>
    <ol>
<li>
  <code>
        ❌
  </code>
<b>
        Create or update a simple gallery.: 0 Fatals, 2 Errors
</b>
</li>
<ul>
  <li>
      Duration:
      0 seconds
    </li>
    <li>
      Example:
      <a href="">
        Microsoft.Compute/preview/2020-09-30/test-scenarios/examples/CreateOrUpdateASimpleGallery.json
      </a>
    </li>
    <li>
      Correlation Id:
      bdc3b522-e988-4892-9cee-26c5a05ac2e7
    </li>
  </ul>
  <br />

| Rule | Message |
|------|---------|

| <code> ❌ </code> Error RESPONSE_INCORRECT_VALUE | /200/body/properties/provisioningState The actual response value is different from example. Path: /200/body/properties/provisioningState. Expected: Updating. Actual: Succeeded |
| <code> ❌ </code> Error RESPONSE_ADDITIONAL_VALUE | /200/body/type Return additional response value. Path: /200/body/type. Expected: undefined. Actual: Microsoft.Compute/galleries |
<li>
  <code>
        ️✔️
  </code>
<b>
        Delete a gallery.: 0 Fatals, 0 Errors
</b>
</li>
<ul>
  <li>
      Duration:
      0 seconds
    </li>
    <li>
      Example:
      <a href="">
        Microsoft.Compute/preview/2020-09-30/test-scenarios/examples/DeleteAGallery.json
      </a>
    </li>
    <li>
      Correlation Id:
      aa44554a-45e9-4677-bc8d-22477762698e
    </li>
  </ul>
  <br />



</ol>
</details>

  <details>
    <summary>
      <code>
        ❌
      </code>
      <b>
        galleries_1: 1 Fatals, 0 Errors
      </b>
    </summary>
    <br />
    <ul>
      <li>
        Duration:
        1 minute, 20.75 seconds
      </li>
      <li>
        Swagger Files
        <ul>
            <li>
              <a href="Microsoft.Compute/preview/2020-09-30/gallery.json">
                Microsoft.Compute/preview/2020-09-30/gallery.json
              </a>
            </li>
        </ul>
      </li>
      <li>
        Run Id: 202106021125-3l03r
      </li>
    </ul>
    <b>
      API Test Steps:
    </b>
    <ol>
<li>
  <code>
        ❌
  </code>
<b>
        Create or update a simple gallery with sharing profile.: 1 Fatals, 0 Errors
</b>
</li>
<ul>
  <li>
      Duration:
      0 seconds
    </li>
    <li>
      Example:
      <a href="">
        Microsoft.Compute/preview/2020-09-30/test-scenarios/examples/CreateOrUpdateASimpleGalleryWithSharingProfile.json
      </a>
    </li>
    <li>
      Correlation Id:
      bd4dfa7c-eabd-47d2-9663-e28f3cc28b57
    </li>
  </ul>
  <br />

| Rule | Message |
|------|---------|
| <code> ❌ </code> Error RUNTIME_ERROR_400 | code: BadRequest, message: Subscription 'db5eb68e-73e2-4fa8-b18a-46cd1be4cce5' is not registered with the feature Microsoft.Compute/SIGSharing. Please register the subscription before retrying the call. |

<li>
  <code>
        ️✔️
  </code>
<b>
        Delete a gallery.: 0 Fatals, 0 Errors
</b>
</li>
<ul>
  <li>
      Duration:
      0 seconds
    </li>
    <li>
      Example:
      <a href="">
        Microsoft.Compute/preview/2020-09-30/test-scenarios/examples/DeleteAGallery.json
      </a>
    </li>
    <li>
      Correlation Id:
      55d776ca-c13d-430d-92b5-c31cb414ab80
    </li>
  </ul>
  <br />



</ol>
</details>

**Comment by ruowan** [comment]
The CI failed, could you please fix the CI.

File: lib/testScenario/postmanCollectionGenerator.ts [hunk 1]
```diff
@@ -49,6 +54,12 @@ export class PostmanCollectionGenerator {
     }
     let index = 0;
     const runId = generateRunId();
+    if (this.opt.markdownReportPath) {
```
Comment (ruowan) [inline lib/testScenario/postmanCollectionGenerator.ts:57]: markdownReportPath is required?

Comment (ruowan) [reply lib/testScenario/postmanCollectionGenerator.ts:57]: better to set default markdownReportPath path in outputFolder.

Comment (zhenglaizhang) [reply lib/testScenario/postmanCollectionGenerator.ts:57]: marked the param as optional

**Comment by zhenglaizhang** [comment]
> The CI failed, could you please fix the CI.

@ruowan fixed, pls help review again.

**Review by ruowan — APPROVED** [review]

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
Please add change log and bump the version.

**Comment by zhenglaizhang** [comment]
> Please add change log and bump the version.

good catch, thanks @raych1 , synced with @ruowan and he will help do this along with his change.

**Pull request merged**